src/main/js/encounterDrawer.js

```diff
@@ -87,9 +88,36 @@ class EncounterDrawerUnstyled extends React.Component {
     this.refreshCombatantsState();
   }
 
-  updateCombatant(combatant) {
-    //TODO patch call
-    this.refreshCombatantsState();
+  updateCombatant(url, data) {
+      fetch(url, {
+          method: 'PATCH',
+          body: JSON.stringify(data),
+          headers: {
+            'Content-Type': 'application/json'
+          }
+      }).catch(err => err)
+      .then(() => this.refreshCombatantsState());
+  }
+
+  handleChange(combatantId, dataType, value) {
+    let data;
+    const newState = this.state.combatants.map(item => {
+        if (item.id == combatantId) {
+            data = {[dataType]: value};
+            return {...item, [dataType]: value};
+        }
+        return item;
+    });
+
+    this.setState({
+       combatants: newState
+    });
+
+    if (!isNaN(value))
+    {
+        let url = 'http://localhost:8080/combatants/' + combatantId;
```

Comment:
Let's leave the url building in the updateCombatant so all the modification methods hide the url.